### PR TITLE
Add shared-project compatibility

### PR DESCRIPTION
**Pull request by mevaere**
User need to add his user_id in the. env file.

### CODE REVIEW

**Review by radeklat — COMMENTED** [review]
Looks sensible. Thank you for your PR and I admire you made sense of this messy code. I've made a few suggestions. Hopefully I allowed running test from forks so they should run on the next change you make.

File: src/config.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ class Config:
         env_file_encoding = "utf-8"
 
     todoist_api_key: str = Field(..., description='See https://todoist.com/prefs/integrations under "API token".')
+    todoist_user_id: int = Field(..., description='Todoist user_id')
```
[the remark this answers is not in the document]
Comment (mevaere) [reply src/config.py:24]: You are right !

File: src/models/todoist_task.py [hunk 2]
```diff
@@ -10,7 +10,9 @@ class TodoistTask(BaseModel):
     content: str
     due: Optional[Dict[str, Any]]  # date, is_recurring, lang, string, timezone
     id: int
+    user_id: int
     is_deleted: int
+    responsible_uid: int
```
[the remark this answers is not in the document]
Comment (mevaere) [reply src/models/todoist_task.py]: Nice ;)

File: src/main.py [hunk 3]
```diff
@@ -104,12 +104,16 @@ def _next_tasks_state_based_on_todoist(self):
...
                 )
-                self._log.info(f"New task {generic_task.content}, {generic_task.state.name}")
+                self._log.info(f"New task {generic_task.content}, {generic_task.state.name}, user_id : {generic_task.user_id},")
 
             self._task_cache.save_task(generic_task)
 
     @staticmethod
     def _should_task_score_points(todoist_task: TodoistTask, generic_task: GenericTask = None) -> bool:
+
+        if (todoist_task.responsible_uid!=-1) and (todoist_task.responsible_uid != get_settings().habitica_user_id):
```
[the remark this answers is not in the document]
Comment (mevaere) [reply src/main.py]: Why not 
`if todoist_task.responsible_uid and (todoist_task.responsible_uid != get_settings().habitica_user_id):`

[the remark this answers is not in the document]
Comment (mevaere) [reply src/main.py]: Error on my side :

`if (todoist_task.responsible_uid is not None) and (todoist_task.responsible_uid != get_settings().todoist_user_id):`

❌ **todoist_user_id** and not **habitica_user_id**

**Pull request closed without merging**